src/huggingface_hub/keras_mixin.py

```diff
@@ -15,6 +19,128 @@
 
 
 logger = logging.getLogger(__name__)
+if is_tf_available():
+    import tensorflow as tf
+
+
+def _extract_hyperparameters_from_keras(model):
+    hyperparameters = dict()
+    if hasattr(model, "optimizer"):
+        hyperparameters["optimizer"] = model.optimizer.get_config()
+    else:
+        hyperparameters["optimizer"] = None
+    hyperparameters[
+        "training_precision"
+    ] = tf.keras.mixed_precision.global_policy().name
+
+    return hyperparameters
+
+
+def _parse_model_history(model):
+    logs = model.history.history
+    num_epochs = len(logs["loss"])
+    lines = []
+
+    for value in range(num_epochs):
+        epoch_dict = {
+            log_key: log_value_list[value] for log_key, log_value_list in logs.items()
+        }
+        values = dict()
+        for k, v in epoch_dict.items():
+            if k.startswith("val_"):
+                k = "validation_" + k[4:]
+            elif k != "epoch":
+                k = "train_" + k
+            splits = k.split("_")
+            name = " ".join([part.capitalize() for part in splits])
+            values[name] = v
+        lines.append(values)
+    return lines
+
+
+def _plot_network(model, save_directory):
+    tf.keras.utils.plot_model(
+        model,
+        to_file=f"{save_directory}/model.png",
+        show_shapes=False,
+        show_dtype=False,
+        show_layer_names=True,
+        rankdir="TB",
+        expand_nested=False,
+        dpi=96,
+        layer_range=None,
+        show_layer_activations=True,
+    )
+
+
+def _write_metrics(lines, model_card):
+    if lines is not None:
+        model_card += "\n| Epochs |"
+
+        for i in lines[0].keys():
+            model_card += f" {i} |"
+        model_card += "\n |"
+        for i in range(len(lines[0].keys()) + 1):
+            model_card += "--- |"  # add header of table
+        for line in lines:
+            model_card += f"\n| {lines.index(line) + 1}|"  # add values
+            for key in line:
+                value = round(line[key], 3)
+                model_card += f" {value}| "
+    else:
+        model_card += "Model history needed"
+    return model_card
+
+
+def _create_model_card(
+    model,
+    repo_dir: Path,
+    model_plot: Optional[bool] = True,
+    task_name: Optional[str] = None,
+):
+    """
+    Creates a model card for the repository.
+    """
+    hyperparameters = _extract_hyperparameters_from_keras(model)
+    if model_plot and is_graphviz_available() and is_pydot_available():
+        _plot_network(model, repo_dir)
+    lines = _parse_model_history(model)
```

Comment:
Hehe lines move all around :smile: 

Try to just define things where they are used. `lines` is not used anywhere in `_create_model_card` except in `_write_metrics`. If `lines` is just used there, why not just create variables directly within the function in which they are used? That way it's also easier to track what `lines` is and future Merve will appreciate when looking through the code.